File: modules/angular2/src/router/lifecycle/route_lifecycle_reflector.dart
library angular.router.route_lifecycle_reflector;

import 'package:angular2/src/router/lifecycle/lifecycle_annotations_impl.dart';
import 'package:angular2/src/router/interfaces.dart';

bool hasLifecycleHook(RouteLifecycleHook e, instance) {
  if (e == routerOnActivate) {
    return instance is OnActivate;
  } else if (e == routerOnDeactivate) {
    return instance is OnDeactivate;
  } else if (e == routerOnReuse) {
    return instance is OnReuse;
  } else if (e == routerCanDeactivate) {
    return instance is CanDeactivate;
  } else if (e == routerCanReuse) {
    return instance is CanReuse;
  }
  return false;
}

File: modules/angular2/src/router/lifecycle/route_lifecycle_reflector.ts
import {Type, isPresent} from 'angular2/src/facade/lang';
import {RouteLifecycleHook} from './lifecycle_annotations_impl';
import {reflector} from 'angular2/src/core/reflection/reflection';

export function hasLifecycleHook(e: RouteLifecycleHook, instance): boolean {
  return e.name in instance;
}
